Source/messaging/MessageClient.cpp

```diff
@@ -138,16 +138,33 @@ namespace Messaging {
     }
 
     /**
-     * @brief Get list of currently announced message controls
+     * @brief Get list of currently announced message modules
      */
-    void MessageClient::Controls(Messaging::MessageUnit::Iterator& controls) const
+    void MessageClient::Modules(std::vector<string>& modules) const
+    {
+        _adminLock.Lock();
+
+        for (auto& client : _clients) {
+            client.second.Modules(modules);
+        }
+
+        std::sort(modules.begin(), modules.end());
```

Comment:
Why is the sorting and the filtering not part of the function client.second.Modules() in line 148 ?

Reply:
The idea behind it was to only sort and remove duplicates once, since we need to do this after all clients have been already asked for their modules. This way we could avoid looking through the modules vector in each client.

But I do agree that from the memory perspective it is much more efficient not to add duplicates at all rather than removing them later on, and I suppose we care more about the used memory, so I'll change it and add a logic to Modules() in Client 😄